libass/ass_render.c

```diff
@@ -863,6 +863,7 @@ void reset_render_context(ASS_Renderer *render_priv, ASS_Style *style)
     render_priv->state.hspacing = style->Spacing;
     render_priv->state.be = 0;
     render_priv->state.blur = style->Blur;
+    render_priv->state.justify = style->Justify;
```

Comment:
I’ve just looked again at where this code is and realized that we don’t actually want to reset `justify` on every `\r`, i. e. in `reset_render_context`. It’s a property of the whole event rather than of individual glyphs, so it doesn’t make sense to change it mid-event, and I think we should set it once per event in `init_render_context` like we set `alignment`. I should’ve spotted this the first time ’round, sorry.